service/controller/resource/eniconfigcrs/create.go

```diff
@@ -2,8 +2,111 @@ package eniconfigcrs
 
 import (
 	"context"
+
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/apis/crd/v1alpha1"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/giantswarm/aws-operator/pkg/annotation"
+	"github.com/giantswarm/aws-operator/pkg/awstags"
+	"github.com/giantswarm/aws-operator/pkg/label"
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if cc.Client.TenantCluster.K8s == nil {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "kubernetes clients are not available in controller context yet")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+		return nil
+	}
+
+	// get security group ID
+	var securityGroupID string
+	{
+		securityGroupID = idFromGroups(cc.Status.TenantCluster.TCCP.SecurityGroups, key.SecurityGroupName(&cr, "aws-cni"))
+	}
+
+	// compute desired ENIConfig CRs
+	var eniConfigs []*v1alpha1.ENIConfig
+	for _, az := range cc.Spec.TenantCluster.TCCP.AvailabilityZones {
+		ec := &v1alpha1.ENIConfig{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: v1alpha1.SchemeGroupVersion.String(),
+				Kind:       "ENIConfig",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					annotation.Docs: "https://godoc.org/github.com/aws/amazon-vpc-cni-k8s/pkg/apis/crd/v1alpha1#ENIConfig",
+				},
+				Labels: map[string]string{
+					label.Cluster:      key.ClusterID(&cr),
+					label.Organization: key.OrganizationID(&cr),
+				},
+				Name:      az.Name,
+				Namespace: corev1.NamespaceDefault,
```

Comment:
Is this the right namespace?

Reply:
Reading the CRD it's a ClusterResource:

```
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: eniconfigs.crd.k8s.amazonaws.com
spec:
  scope: Cluster
  group: crd.k8s.amazonaws.com
  version: v1alpha1
  names:
    plural: eniconfigs
    singular: eniconfig
    kind: ENIConfig
```

Here is were the CNI will try to read from https://github.com/aws/amazon-vpc-cni-k8s/blob/b0286f121f6996fbdc06f701c31b908bc214c2fa/pkg/eniconfig/eniconfig.go#L156

```
// Start kicks off ENIConfig controller
func (eniCfg *ENIConfigController) Start() {
	printVersion()

	sdk.ExposeMetricsPort()

	resource := "crd.k8s.amazonaws.com/v1alpha1"
	kind := "ENIConfig"
	resyncPeriod := time.Second * 5
	log.Infof("Watching %s, %s, every %v s", resource, kind, resyncPeriod.Seconds())
	sdk.Watch(resource, kind, "", resyncPeriod)
	sdk.Watch("/v1", "Node", corev1.NamespaceAll, resyncPeriod)
	sdk.Handle(NewHandler(eniCfg))
	sdk.Run(context.TODO())
}
```

Reply:
so not sure if the namespace thing will work or we need to handle this differently

Reply:
the documentation and example show ENiConfig in the default namespace, and there is no mention of the namespace in the `aws-cni` config so I assume `default` ns is the right one.

If that doesn't work we might try `kube-system` as that's the one where `aws-cni` pod runs.